packages/property-view/src/browser/resource-property-view/resource-property-data-service.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+disableJSDOM();
+
+let resourcePropertyDataService: ResourcePropertyDataService;
+
+const mockFileStat: FileStat = {
+    isFile: false,
+    isDirectory: true,
+    isSymbolicLink: false,
+    resource: new URI('resource'),
+    name: 'name'
+};
+
+describe('resource-property-data-service', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertyDataService).toSelf().inSingletonScope();
+        container.bind(FileService).toConstantValue({
+            async resolve(uri: URI): Promise<FileStat> {
+                return mockFileStat;
+            }
+        } as FileService);
+        container.bind(PropertyDataService).to(ResourcePropertyDataService).inSingletonScope();
+        resourcePropertyDataService = container.get(ResourcePropertyDataService);
+    });
+
+    after(() => {
+        disableJSDOM();
+    });
+
+    const fileSelection: FileSelection = {
+        fileStat: mockFileStat
+    };
+
+    const navigatableSelection: Navigatable = {
+        getResourceUri(): URI | undefined {
+            return new URI('resource-uri');
+        },
+        createMoveToUri(): URI | undefined {
+            return new URI('move-uri');
+        }
+    };
+
+    const arrayFileSelection: FileSelection[] = [fileSelection, { fileStat: { ...mockFileStat, resource: new URI('secondURI') } }];
+
+    describe('#canHandle', () => {
+
+        it('should handle an empty object', () => {
+            expect(resourcePropertyDataService.canHandleSelection({})).eq(0);
+        });
+
+        it('should handle undefined', () => {
+            expect(resourcePropertyDataService.canHandleSelection(undefined)).eq(0);
+        });
+
+        it('should handle a file selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(arrayFileSelection)).to.be.greaterThan(0);
+        });
+
+        it('should handle a navigatable selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(navigatableSelection)).to.be.greaterThan(0);
+        });
+    });
+
+    describe('#providePropertyData', () => {
+
+        it('should return the stat of a file selection', async () => {
+            const data = await resourcePropertyDataService.providePropertyData(arrayFileSelection);
+            expect(data).to.equal(mockFileStat);
+        });
+
+        it('should return the first `FileStat` for multiple file selections', async () => {
+            const data = await resourcePropertyDataService.providePropertyData(arrayFileSelection);
+            expect(data).to.equal(arrayFileSelection[0].fileStat);
+        });
+
+        it('should return undefined if the selection is undefined', async () => {
+            const data = await resourcePropertyDataService.providePropertyData(undefined);
+            expect(data).to.equal(undefined);
+        });
```

Comment:
This test should probably got last in the group:

- test file selection
- test multiple file selection
- test navigatable selection
- test undefined